app/services/user-agent-service/datomstore.js

```diff
@@ -0,0 +1,239 @@
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+
+import Immutable from 'immutable';
+import microtime from 'microtime-fast';
+import { Bookmark } from '../../shared/model';
+
+import { SessionEndReason, SessionStartReason, SnippetSize, StarOp, VisitType } from './storage';
+
+import {
+  datascript,
+  mori,
+  helpers,
+} from 'datascript-mori';
+
+const { hashMap, vector, parse, toJs, equals, isMap, hasKey, isSet, set, getIn, get } = mori;
+const { DB_ADD } = helpers;
+
+export class ProfileDatomStorage {
+  constructor(dir) {
+    this.dir = dir;
+    const schema = this.createSchema();
+    this.conn = datascript.core.create_conn(schema);
+    this.schema = schema;
+  }
+
+  createSchema() {
+    return helpers.schema_to_clj({
+      'page/url': {
+        ':db/cardinality': ':db.cardinality/one',
+        ':db/unique': ':db.unique/identity',
+        ':db/doc': 'A page\'s URL.',
+        ':db.install/_attribute': ':db.part/db',
+      },
+
+      'page/title': {
+        ':db/ident': ':page/title',
+        ':db/cardinality': ':db.cardinality/one',      // We supersede as we see new titles.
+        ':db/doc': 'A page\'s title.',
+        ':db.install/_attribute': ':db.part/db',
+      },
+
+      'event/visit': {
+        ':db/valueType': ':db.type/ref',
+        ':db/cardinality': 'db.cardinality/many',
```

Comment:
This doesn't actual work yet, so don't land it.  Specifically, you have `'db.cardinality/many'` when you need `':db.cardinality/many'` (observe colon).